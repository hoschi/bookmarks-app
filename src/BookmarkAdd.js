import React from 'react'
import { Mutation } from 'react-apollo'
import BookmarkAddForm from './BookmarkAddForm'
import { createBookmark, queryAllBookmarks } from './gql'

function BookmarkAdd() {
    return (
        <Mutation
            mutation={createBookmark}
            ignoreResults
            refetchQueries={[{ query: queryAllBookmarks }]}
        >
            {(createBookmark) => {
                return <BookmarkAddForm createBookmark={createBookmark} />
            }}
        </Mutation>
    )
}

export default BookmarkAdd
