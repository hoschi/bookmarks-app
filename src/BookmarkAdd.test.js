import React from 'react'
import { shallow } from 'enzyme'
import { Mutation } from 'react-apollo'
import BookmarkAdd from './BookmarkAdd'

test('render', function() {
    const actual = shallow(<BookmarkAdd />)
    expect(actual).toMatchSnapshot()
})

describe('createBookmark Mutation', function() {
    const actual = shallow(<BookmarkAdd />)
        .find(Mutation)
        .prop('children')

    const createBookmark = jest.fn()
    const bookmarkAddForm = shallow(actual(createBookmark))
    expect(bookmarkAddForm).toMatchSnapshot()
    expect(bookmarkAddForm.prop('createBookmark')).toBe(createBookmark)
})
