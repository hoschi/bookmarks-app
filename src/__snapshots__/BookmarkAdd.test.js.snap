// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<Formik
  createBookmark={[MockFunction]}
  enableReinitialize={false}
  handleSubmit={[Function]}
  initialValues={Object {}}
  isInitialValid={false}
  onSubmit={[Function]}
  render={[Function]}
  validate={[Function]}
  validateOnBlur={true}
  validateOnChange={true}
/>
`;

exports[`render 1`] = `
<Mutation
  ignoreResults={true}
  mutation={
    Object {
      "definitions": Array [
        Object {
          "directives": Array [],
          "kind": "OperationDefinition",
          "name": Object {
            "kind": "Name",
            "value": "CreateBookmark",
          },
          "operation": "mutation",
          "selectionSet": Object {
            "kind": "SelectionSet",
            "selections": Array [
              Object {
                "arguments": Array [
                  Object {
                    "kind": "Argument",
                    "name": Object {
                      "kind": "Name",
                      "value": "id",
                    },
                    "value": Object {
                      "kind": "Variable",
                      "name": Object {
                        "kind": "Name",
                        "value": "id",
                      },
                    },
                  },
                  Object {
                    "kind": "Argument",
                    "name": Object {
                      "kind": "Name",
                      "value": "isRead",
                    },
                    "value": Object {
                      "kind": "BooleanValue",
                      "value": false,
                    },
                  },
                  Object {
                    "kind": "Argument",
                    "name": Object {
                      "kind": "Name",
                      "value": "title",
                    },
                    "value": Object {
                      "kind": "Variable",
                      "name": Object {
                        "kind": "Name",
                        "value": "title",
                      },
                    },
                  },
                  Object {
                    "kind": "Argument",
                    "name": Object {
                      "kind": "Name",
                      "value": "url",
                    },
                    "value": Object {
                      "kind": "Variable",
                      "name": Object {
                        "kind": "Name",
                        "value": "url",
                      },
                    },
                  },
                ],
                "directives": Array [],
                "kind": "Field",
                "name": Object {
                  "kind": "Name",
                  "value": "createBookmark",
                },
                "selectionSet": Object {
                  "kind": "SelectionSet",
                  "selections": Array [
                    Object {
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "id",
                      },
                    },
                  ],
                },
              },
            ],
          },
          "variableDefinitions": Array [
            Object {
              "kind": "VariableDefinition",
              "type": Object {
                "kind": "NonNullType",
                "type": Object {
                  "kind": "NamedType",
                  "name": Object {
                    "kind": "Name",
                    "value": "ID",
                  },
                },
              },
              "variable": Object {
                "kind": "Variable",
                "name": Object {
                  "kind": "Name",
                  "value": "id",
                },
              },
            },
            Object {
              "kind": "VariableDefinition",
              "type": Object {
                "kind": "NonNullType",
                "type": Object {
                  "kind": "NamedType",
                  "name": Object {
                    "kind": "Name",
                    "value": "String",
                  },
                },
              },
              "variable": Object {
                "kind": "Variable",
                "name": Object {
                  "kind": "Name",
                  "value": "title",
                },
              },
            },
            Object {
              "kind": "VariableDefinition",
              "type": Object {
                "kind": "NonNullType",
                "type": Object {
                  "kind": "NamedType",
                  "name": Object {
                    "kind": "Name",
                    "value": "String",
                  },
                },
              },
              "variable": Object {
                "kind": "Variable",
                "name": Object {
                  "kind": "Name",
                  "value": "url",
                },
              },
            },
          ],
        },
      ],
      "kind": "Document",
      "loc": Object {
        "end": 179,
        "source": Object {
          "body": "
    mutation CreateBookmark($id: ID!, $title: String!, $url: String!) {
        createBookmark(id: $id, isRead: false, title: $title, url: $url) {
            id
        }
    }
",
          "locationOffset": Object {
            "column": 1,
            "line": 1,
          },
          "name": "GraphQL request",
        },
        "start": 0,
      },
    }
  }
  refetchQueries={
    Array [
      Object {
        "query": Object {
          "definitions": Array [
            Object {
              "directives": Array [],
              "kind": "OperationDefinition",
              "operation": "query",
              "selectionSet": Object {
                "kind": "SelectionSet",
                "selections": Array [
                  Object {
                    "arguments": Array [
                      Object {
                        "kind": "Argument",
                        "name": Object {
                          "kind": "Name",
                          "value": "sortField",
                        },
                        "value": Object {
                          "block": false,
                          "kind": "StringValue",
                          "value": "isRead",
                        },
                      },
                      Object {
                        "kind": "Argument",
                        "name": Object {
                          "kind": "Name",
                          "value": "sortOrder",
                        },
                        "value": Object {
                          "block": false,
                          "kind": "StringValue",
                          "value": "ASC",
                        },
                      },
                    ],
                    "directives": Array [],
                    "kind": "Field",
                    "name": Object {
                      "kind": "Name",
                      "value": "allBookmarks",
                    },
                    "selectionSet": Object {
                      "kind": "SelectionSet",
                      "selections": Array [
                        Object {
                          "arguments": Array [],
                          "directives": Array [],
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "id",
                          },
                        },
                        Object {
                          "arguments": Array [],
                          "directives": Array [],
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "title",
                          },
                        },
                        Object {
                          "arguments": Array [],
                          "directives": Array [],
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "url",
                          },
                        },
                        Object {
                          "arguments": Array [],
                          "directives": Array [],
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "isRead",
                          },
                        },
                      ],
                    },
                  },
                ],
              },
              "variableDefinitions": Array [],
            },
          ],
          "kind": "Document",
          "loc": Object {
            "end": 153,
            "source": Object {
              "body": "
    {
        allBookmarks(sortField: \\"isRead\\", sortOrder: \\"ASC\\") {
            id
            title
            url
            isRead
        }
    }
",
              "locationOffset": Object {
                "column": 1,
                "line": 1,
              },
              "name": "GraphQL request",
            },
            "start": 0,
          },
        },
      },
    ]
  }
>
  <Component />
</Mutation>
`;
